asmcomp/arm/emit.mlp

```diff
@@ -798,26 +798,48 @@ let emit_instr i =
 
 (* Emission of an instruction sequence *)
 
+(* consider the degenerate case where a single literal is followed by
+   a jump table longer than 4KB; we have to emit the constant pool
+   before the jump table or it will be too late *)
+let emit_instr_before_pool i =
+  match i.desc with
+  | Lswitch jumptbl -> 2 + Array.length jumptbl
+  | _ -> emit_instr i
```

Comment:
It seems fragile to me to have three repeated matches all of which have to be in sync.  How about changing `emit_instr_before_pool` to be called `emit_or_defer_instr` and return the size together with either `None` (no instruction deferred) or `Some instr` (for `Lswitch`).  Then instead of changing `fallthrough`, which seems a bit misleading, you could just test for `Some` and avoid the third match; you just emit the instruction in the `Some`.  It might not be worth having `emit_instr_after_pool` at that point.